### DISCUSSION TITLE
Route duration calculation

### DISCUSSION THREAD

**Discussion by bios4** [devemux86/cruiser#481] [Q&A] [answered]
Today I dealt with the topic of route duration in cruiser for the first time.
I used my ride home today to test it - a distance of approx. 121 km...

With the setting "Route duration - Calculated route" the result is a duration of 8:50 hours, which is absolute nonsense.

<img src="https://github.com/devemux86/cruiser/assets/145460147/e13a1430-3775-469c-bc9b-5d82d557eaf1" width="33%">
<img src="https://github.com/devemux86/cruiser/assets/145460147/cbb2adc5-3c00-495a-9fed-b4c48fffda5a" width="33%">

If you set the setting to "Route duration - Average speed", Cruiser calculates a duration of 2:41 hours, which is about right for the selected route.

<img src="https://github.com/devemux86/cruiser/assets/145460147/50844162-1ffd-4935-8642-3a300ca640f8" width="33%">
<img src="https://github.com/devemux86/cruiser/assets/145460147/223e5626-91bc-40d8-9f53-9f21ea8b0ffc" width="33%">

My question now is: what data exactly does Cruiser use to calculate the route duration when the setting is set to "Calculated route"?

**Comment by devemux86** [comment]
@bios4 thanks for the report.

"Calculated route" option uses the route duration that exists in the route response from the router.
Each router calculates internally the route duration and differently based on its own speed selection.

Can you provide the route and profile to check it?

**Reply by bios4** [reply]
> Can you provide the route and profile to check it?

I will send it to you per E-mail as it contains private data/locations...

**Reply by devemux86** [reply]
Remember that if it is routing with BRouter (offline), we can also check the result and duration in [BRouter website](https://brouter.de/brouter-web/).

**Comment by devemux86** [comment]
It seems that BRouter - as we avoid more and more roads - adjusts the route duration.
Using smaller roads with smaller speeds cause a larger route duration.

This can also be tested on [BRouter website](https://brouter.de/brouter-web/): load route as gpx and use driving-cruiser profile.
- Travel time with default settings is 3:03 h
- Avoid motorways / trunk / primary 5:02 h
- Use town / forest options 6:44 h

Probably we need to only use the average speed in BRouter calculations.

cc @afischerdev

**Comment by devemux86** [comment]
From the next Cruiser, the average speed will be the default in route duration, so users can get more personalized information:
- https://github.com/devemux86/cruiser/discussions/482

**Comment by devemux86** [comment, marked as answer]
Implemented in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
